### PR TITLE
[JENKINS-31832] fixed job class hierarchy

### PR DESCRIPTION
**Pull request by daspilker**

### CODE REVIEW

**Comment by TheKrush** [comment]
This pull request broke authenticationToken (which is still functional in jenkins). https://groups.google.com/forum/#!topic/job-dsl-plugin/wQpbhPrafHY

**Comment by mac-110** [comment]
Why the hell you removed the authjenticationToken? It is still avaible in jenkins pipeline job type?

**Comment by sunsear** [comment]
Please restore the authenticationToken functionality in the jobDSL.

File: docs/Migration.md [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+    jdk('jdk-1.8')
+    label('linux')
+    lockableResources('lock-resource')
+    multiscm {
+         // ...
+    }
+    publishers {
+         // ...
+    }
+    scm {
```
[the remark this answers is not in the document]
Comment (jglick) [reply docs/Migration.md]: I do not know what `scm` would be in this context. You will need to ask @daspilker, the maintainer of this plugin.

**Pull request merged**